File: src/python/t896_MonotonicArray.py
def isMonotonic(self, nums: List[int]) -> bool:
    if len(nums) == 1:
        return True
    i = 1
    while i < len(nums) - 1:
        if nums[i] != nums[i - 1]:
            break
        i = i + 1
    if nums[i] > nums[i - 1]:
        for j in range(i, len(nums)):
            if nums[j] < nums[j - 1]:
                return False
    else:
        for j in range(i, len(nums)):
            if nums[j] > nums[j - 1]:
                return False
    return True

File: src/t896_MonotonicArray.java
public class t896_MonotonicArray {
    public boolean isMonotonic(int[] arr) {
        if (arr.length == 1)
            return true;
        int i = 1;
        while (i < arr.length - 1) {
            if (arr[i] != arr[i - 1])
                break;
            i++;
        }
        if (arr[i] > arr[i - 1]) {
            for (int j = i; j < arr.length; j++) {
                if (arr[j] < arr[j - 1]) return false;
            }
        } else {
            for (int j = i; j < arr.length; j++) {
                if (arr[j] > arr[j - 1]) return false;
            }
        }
        return true;
    }
}
